cpp/vm/evmzero/interpreter.cc

```diff
@@ -1328,6 +1328,41 @@ struct Impl<OpCode::JUMPDEST> {
   static OpResult Run() noexcept { return {}; }
 };
 
+template <>
+struct Impl<OpCode::TLOAD> {
+  constexpr static OpInfo kInfo{
+      .pops = 1,
+      .pushes = 1,
+      .static_gas = 100,
+      .introduced_in = EVMC_CANCUN,
+  };
+
+  static OpResult Run(uint256_t* top, Context& ctx) noexcept {
+    const uint256_t key = top[0];
+    evmc::bytes32 value = ctx.host->get_transient_storage(ctx.message->recipient, ToEvmcBytes(key));
+    top[0] = ToUint256(value);
+    return {};
+  }
+};
+
+template <>
+struct Impl<OpCode::TSTORE> {
+  constexpr static OpInfo kInfo{
+      .pops = 2,
+      .pushes = 0,
+      .static_gas = 100,
+      .disallowed_in_static_call = true,
```

Comment:
I was also surprised, especially since the evm.codes website also doesn't list the static call context as a possible error.
Here is the geth implementation:
https://github.com/ethereum/go-ethereum/blob/86150af2e5deb33130660a5a5d62ed819a2b0e62/core/vm/eips.go#L211
and here is the implementation in the execution specs: (thought this would be our chance to find a difference between geth implementation and their execution spec.
https://github.com/ethereum/execution-specs/blob/c5415056a4a7066906f67c203ec5364a9de8e017/src/ethereum/cancun/vm/instructions/storage.py#L171

Reply:
There are no errors in geth, it is always right.

Reply:
Thanks for checking this and finding the evidence!

Reply:
The [EIP-1153](https://eips.ethereum.org/EIPS/eip-1153) explicitly states that TSTORE should not be allowed in static contexts. So it is an error on [evm.codes](https://www.evm.codes/?fork=cancun).